src/components/Settings/SettingsDialog.tsx

```diff
@@ -1,23 +1,36 @@
 import { css } from '@emotion/react'
-import { Box, Button, Dialog, Flex, ScrollArea } from '@radix-ui/themes'
+import { Box, Button, Dialog, Flex, ScrollArea, Tabs } from '@radix-ui/themes'
 import { ProxySettings } from './ProxySettings'
 import { FormProvider, useForm } from 'react-hook-form'
 import { AppSettingsSchema } from '@/schemas/appSettings'
 import { zodResolver } from '@hookform/resolvers/zod'
 import { useEffect, useState } from 'react'
-import { ButtonWithTooltip } from '@/components/ButtonWithTooltip'
 import { RecorderSettings } from './RecorderSettings'
 import { AppSettings } from '@/types/settings'
 import { UsageReportSettings } from './UsageReportSettings'
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons'
+import { ButtonWithTooltip } from '../ButtonWithTooltip'
+import { findIndex, sortBy } from 'lodash-es'
 
 type SettingsDialogProps = {
   open: boolean
   onOpenChange: (open: boolean) => void
 }
 
+const tabs = [
```

Comment:
I extracted the content of the tabs to an array, so we can loop through and select which one to focus on after an error happens, following the order they are rendered in the UI. This ensures that the user is taken to the very first error at the top of the form. 

This loop happens in the `onInvalid` method on L72.

Reply:
> so we can loop through and select which one to focus on after an error happens

Looks like saving on blur/change can help us avoid the problem altogether: we won't need to show the warning icon and focus on a broken tab, and the user won't need to remember to save when the changes they made are on a different tab